FluentTerminal.SystemTray/Utilities.cs

```diff
@@ -569,44 +569,58 @@ internal static void SaveFile(string path, string content)
         {
-            bool? deviceMuted = VolumeControl.GetDefaultAudioEndpointMute();
-            if (deviceMuted != true)
-            {
-                VolumeControl.SetDefaultAudioEndpointMute(true);
-            }
-
-            Process cmdProcess = new Process();
-            cmdProcess.StartInfo.FileName = "cmd.exe";
-            cmdProcess.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
-            cmdProcess.StartInfo.Arguments = "/k \x07";
-            cmdProcess.Start();
-
-            bool wasConhostMuted = false;
+            bool audioSessionFound = false;
             int attempt = 0;
-            while (wasConhostMuted == false && attempt++ < 2 && !cmdProcess.HasExited)
+            while (audioSessionFound == false && attempt++ < attemptCount)
             {
-                cmdProcess.WaitForExit(1500);
-
                 foreach (Process conhost in Process.GetProcessesByName("conhost"))
                 {
                     try
                     {
-                        bool? isMuted = null;
-                        isMuted = VolumeControl.GetAudioSessionMute(conhost.Id);
-                        if (isMuted != null && isMuted != mute)
+                        bool? isAudioSessionMuted = null;
+                        isAudioSessionMuted = VolumeControl.GetAudioSessionMute(conhost.Id);
+                        if (isAudioSessionMuted != null && isAudioSessionMuted != mute)
                         {
                             VolumeControl.SetAudioSessionMute(conhost.Id, mute);
-                            wasConhostMuted = true;
                             Logger.Instance.Debug($"Mute state for conhost process with id={conhost.Id} was set to {mute}.");
                         }
+
+                        if (isAudioSessionMuted != null)
+                        {
+                            audioSessionFound = true;
+                        }
                     }
                     catch (Exception) { }
                 }
             }
+            return audioSessionFound;
+        }
+
+        internal static void MuteTerminal(bool mute)
+        {
+            bool audioSessionFound = MuteConhost(mute, 1);
+
+            if (audioSessionFound == true)
+                return;
+
+            bool? deviceMuted = VolumeControl.GetDefaultAudioEndpointMute();
+            if (deviceMuted != true)
+            {
+                VolumeControl.SetDefaultAudioEndpointMute(true);
+            }
+
+            Process cmdProcess = new Process();
+            cmdProcess.StartInfo.FileName = "cmd.exe";
+            cmdProcess.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
+            cmdProcess.StartInfo.Arguments = "/k \x07";
+            cmdProcess.Start();
+
+            cmdProcess.WaitForExit(1500);
+
+            MuteConhost(mute, 2);
 
-            cmdProcess.Kill();
             if (deviceMuted != true)
             {
                 VolumeControl.SetDefaultAudioEndpointMute(false);

```

Comment:
This should probably be a in try-finally to ensure the audio endpoint is always unmuted even if something goes wrong.

Reply:
Have removed muting of device as it was discussed ( in next PR will attempt to mute audio session on its early start )